### PR TITLE
 Try to get hot-reload working on the material-ui demo

### PR DESCRIPTION
**Pull request by AlexITC**
This is just a try as I'm not able to get the hot-reload working properly, I have grabbed the code from the default slinky template but for some reason, I get a blank page.

To be honest, I'm not that familiar with scalajs to understand what I'm missing, hopefully someone around is.

Thanks.

### CODE REVIEW

**Comment by oyvindberg** [comment]
So yeah, I've never really bothered with hot reloading myself. 
I saw you saw https://gitter.im/shadaj/slinky?at=5f78d04e65cb03206069c032 , so I guess it's possible to try to rewrite functional components into class components - I think it's a pretty easy refactoring

**Comment by AlexITC** [comment]
I did, the problem seems to be related to the webpack config, hot-reloading even with functional components should work better that restarting the whole app, I'll give another try soon.

**Comment by AlexITC** [comment]
FYI: I got this working on a private project, I'll try to port it to the examples soon.

**Comment by oyvindberg** [comment]
any luck here @AlexITC ?

**Comment by AlexITC** [comment]
@oyvindberg I forget to upload the changes, here they are, running `sbt dev` gets such hot-reloading effect, which is not the same as how it is when using the non-functional components, but, to me, is far better that manual restart.

File: build.sbt [hunk 1]
```diff
@@ -52,6 +52,7 @@ lazy val baseSettings: Project => Project =
       scalaJSLinkerConfig := scalaJSLinkerConfig.value.withSourceMap(false),
       /* for slinky */
       libraryDependencies ++= Seq("me.shadaj" %%% "slinky-hot" % "0.6.5"),
+      libraryDependencies += "me.shadaj" %%% "slinky-web" % "0.6.5",
```
Comment (oyvindberg) [inline build.sbt:55]: this shouldn't be necessary, or?

File: build.sbt [hunk 2]
```diff
@@ -174,6 +175,7 @@ lazy val `material-ui` =
       webpackDevServerPort := 8008,
       stFlavour := Flavour.Slinky,
       stReactEnableTreeShaking := Selection.All,
+      stIgnore += "react-proxy",
```
Comment (oyvindberg) [inline build.sbt:178]: I know I havent done this, but if you add a comment why it's ignored that would be nice.

File: build.sbt [hunk 3]
```diff
@@ -182,8 +184,10 @@ lazy val `material-ui` =
         "@types/recharts" -> "1.8.10",
         "@types/classnames" -> "2.2.10",
         "react-router-dom" -> "5.1.2",
-        "@types/react-router-dom" -> "5.1.2"
-      )
+        "@types/react-router-dom" -> "5.1.2",
+        "react-proxy" -> "1.1.8"
```
Comment (oyvindberg) [inline build.sbt:188]: what is `react-proxy` for? is it related to `hot`?

File: build.sbt [hunk 4]
```diff
@@ -321,10 +325,8 @@ lazy val withCssLoading: Project => Project =
     )
   )
 
-/**
-  * Implement the `start` and `dist` tasks defined above.
-  * Most of this is really just to copy the index.html file around.
-  */
+addCommandAlias("dev", ";material-ui/fastOptJS::startWebpackDevServer;~material-ui/fastOptJS")
```
Comment (oyvindberg) [inline build.sbt:328]: call it `muiDev` or something, there are many other demos and this will be global

**Review by oyvindberg — APPROVED** [review]
Tried it now, and it was pretty smooth :+1: 

please address the small comments and merge when you're happy.

File: build.sbt (hunk 1 above)
Comment (AlexITC) [reply build.sbt:55]: Indeed, I got this working after several tries

File: build.sbt (hunk 2 above)
Comment (AlexITC) [reply build.sbt:178]: The hot reloading mechanism requires the `react-proxy`, which doesn't have the typescript definitions, without this, the compilation fails, in any case, I'll add a comment.

File: build.sbt (hunk 3 above)
Comment (AlexITC) [reply build.sbt:188]: Yes, otherwise, compilation fails

File: build.sbt (hunk 4 above)
Comment (AlexITC) [reply build.sbt:328]: Yes, in fact, I'd propagate the hot reloading to all the demos, and specify the command `dev` command for each of those, I'll update the PR once I get some free time

**Comment by AlexITC** [comment]
At the end, this took me too long, I fixed the minor details and merged (887b81690693e35189ab199578d6ffe88b863ae6), it would be nice to include hot-reloading on the remaining projects.

**Pull request closed without merging**